src/context.rs

```diff
@@ -352,14 +352,15 @@ impl<'buf> Secp256k1<AllPreallocated<'buf>> {
     /// Creates a context from a raw context.
     ///
     /// # Safety
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
-    /// * The user must handle the freeing of the context(using the correct functions) by himself.
-    /// * Violating these may lead to Undefined Behavior.
     ///
+    /// This is highly unsafe due to the number of conditions that aren't checked, specifically:
+    ///
+    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer that was generated by *exactly*
+    ///   the same code/version of the libsecp256k1 used here.
+    /// * The capabilities (`All`/`SignOnly`/`VerifyOnly`) of the context *must* match the flags
+    ///   passed to libsecp256k1 when generating the context.
+    /// * The user must handle the freeing of the context (using the correct functions) by himself.
+    /// * `raw_ctx` must point to writable memory (cannot be `ffi::secp256k1_context_no_precomp`).
```

Comment:
The lifetime of `raw_ctx` pointer must outlive `'buf`.

Non-safety warning: the returned value will **not** deallocate the pointer, nor destroy the context which may lead to memory leaks. The caller is responsible for these. Note that `ManuallyDrop::drop` (or `drop_in_place`) will **only** destroy the context, it will **not** free memory behind the pointer.

Reply:
It does but claiming it's safety is wrong and it might not be clear how to drop it properly.

Reply:
In 382206f0372130d407c29ca3221878dcfc47495e:

nit: I'd add the words "in particular," before "cannot be", to clarify that *any* non-writeable memory is a problem, not just the static context.